src/language/markdown-source-code.js

```diff
@@ -26,13 +26,13 @@ import { findOffsets } from "../util.js";
 /** @typedef {import("@eslint/core").File} File */
 /** @typedef {import("@eslint/core").TraversalStep} TraversalStep */
 /** @typedef {import("@eslint/core").VisitTraversalStep} VisitTraversalStep */
-/** @typedef {import("@eslint/core").TextSourceCode} TextSourceCode */
+/** @typedef {import("@eslint/core").TextSourceCode<{LangOptions: MarkdownLanguageOptions, RootNode: RootNode, SyntaxElementWithLoc: MarkdownNode, ConfigNode: { value: string; position: SourceLocation }}>} TextSourceCode */
```

Comment:
```suggestion
/** @import { TextSourceCode } from "@eslint/core"; */
```

Maybe we could use an `@import` tag here to import `TextSourceCode` as a generic type and add the type parameters on the annotation of `MarkdownSourceCode` instead? Like it's done in `@eslint/json`:

https://github.com/eslint/json/blob/dce4601b1a40ceaeb4b61fbcbef0170a67b73e37/src/languages/json-source-code.js#L68

Reply:
[JSON](https://github.com/eslint/json/blob/dce4601b1a40ceaeb4b61fbcbef0170a67b73e37/src/languages/json-source-code.js#L22-L27) only uses `@import` and doesn’t use `@typedef`.

```js
/**
 * @import { DocumentNode, Node, Token } from "@humanwhocodes/momoa";
 * @import { SourceLocation, FileProblem, DirectiveType, RulesConfig, TextSourceCode } from "@eslint/core";
 * @import { JSONSyntaxElement } from "../types.ts";
 * @import { JSONLanguageOptions } from "./json-language.js";
 */
```

So, would it be better to convert all `@typedef` to `@import`?

Or should we leave the other `@typedef` statements as they are and only use `@import` for `TextSourceCode`?

Reply:
> So, would it be better to convert all `@typedef` to `@import`?
> 
> Or should we leave the other `@typedef` statements as they are and only use `@import` for `TextSourceCode`?

`@eslint/json` uses additional logic to handle `@import` tags during the build in [tools/dedupe-types.js](https://github.com/eslint/json/blob/main/tools/dedupe-types.js). If we want switch to `@import` everywhere, we should also update the build scripts in this repo, which is probably a good idea but out of scope for this change.

Alternatively, we could keep using the `@typedef` syntax with `@template` to re-export `TextSourceCode` as a generic type, like with `MarkdownRuleDefinition` here:

https://github.com/eslint/markdown/blob/13dfd5ec279d1db3cbd1ef2b6bf5d00799371090/src/index.js#L22-L25

Reply:
I'd like to use `@typedef` and `@template` for this PR.

Once all my current PRs are resolved, I'll open another one to switch to the `@import` syntax!